addons/website_blog/static/src/js/blog_tag_add.js

```diff
@@ -0,0 +1,336 @@
+odoo.define('website_blog.blog_post_tag.add', function (require) {
+'use strict';
+
+const {qweb, _t} = require('web.core');
+const options = require('web_editor.snippets.options');
+const {Class: EditorMenuBar} = require('web_editor.editor');
+
+options.registry.TagBlogPostSelect = options.Class.extend({
+    xmlDependencies: (options.Class.prototype.xmlDependencies || []).concat(['/website_blog/static/src/xml/website_blog_tag.xml']),
+    /**
+     * @override
+     */
+    willStart: async function () {
+        const _super = this._super.bind(this);
+        this.blogPostId = this.$target.data('blog-id');
+        this.addList = [];
+        this.unlinkList = [];
+        this.isEditing = false;
+        this.trigger_up('set_blogPost', {
+            blogPostId: this.blogPostId,
+        });
+        await this._getTags();
+        return _super(...arguments);
+    },
+
+    //--------------------------------------------------------------------------
+    // Options
+    //--------------------------------------------------------------------------
+
+    /**
+     * @see this.selectClass for params
+     */
+    editList: function (previewMode, widgetValue, params) {
+        this.isEditing = true;
+        this._toggleEditUI();
+        this.rerender = true;
+    },
+    /**
+     * @see this.selectClass for params
+     */
+    saveTagList: function (previewMode, widgetValue, params) {
+        this.isEditing = false;
+        this._saveChanges();
+        this._toggleEditUI();
+        this.rerender = true;
+    },
+    /**
+     * @see this.selectClass for params
+     */
+    confirmNew: function (previewMode, widgetValue, params) {
+        const name = $('[data-name="add_tag_opt"] input').val();
+        const existing = this.addList.some(tag => tag.name.toLowerCase() === name.toLowerCase()) || this.tagIds.some(tag => tag.name.toLowerCase() === name.toLowerCase());
+        if (!name) {
+            return;
+        }
+        if (existing) {
+            return this.displayNotification({
+                type: 'warning',
+                message: _t('Tag already exist'),
+            });
+        }
+        const newTag = {
+            'id': 0,
+            'name': name,
+        };
+        this.tagIds.push(newTag);
+        this.addList.push(newTag);
+        this._addTagPreview(newTag);
+        this.trigger_up('set_tagList', {
+            addList: this.addList,
+            unlinkList: this.unlinkList,
+        });
+        this.rerender = true;
+    },
+    /**
+     * @see this.selectClass for params
+     */
+    removeTag: function (previewMode, widgetValue, params) {
```

Comment:
Could you make the UI a bit smarter? Right now you have we-button inside we-button, a "minus" icon which is a decoration as clicking on the whole tag deletes it, but the cursor is only pointer over the "minus" icon... please make it better.